samples/glean/build.gradle

```diff
@@ -43,3 +39,5 @@ dependencies {
 }
 
 apply from: '../../components/service/glean/scripts/sdk_generator.gradle'
+
+ext.setupGlean('./components/service/glean')
```

Comment:
As stated above, this woldn't be possible if the sample app lived outside of this repo. Let's keep the scripts as they are, even if we end up with two conda environments. That's ok, as the library and the app are two distinct things, as if they were different repos.

@pocmo is there something we could do to cache the conda installer/packages to be nicer to them?

Reply:
Would it be possible to use the `buildDir` of `rootProject` to share the Python environment? I'm not sure whether this could cause weird side effects when building parallel though (although some parts of samples-glean needs to wait for services-glean to complete).

Having the environment twice is not the best for saving disk space; however that's how the setup with real apps will be too. So i'm not too concerned.

Reply:
> Would it be possible to use the `buildDir` of `rootProject` to share the Python environment? I'm not sure whether this could cause weird side effects when building parallel though (although some parts of samples-glean needs to wait for services-glean to complete).

Mh, good point. @pocmo, is the following reasoning correct:

- `rootProject.buildDir == buildDir` for most, common apps
- `rootProject.buildDir != buildDir` if glean is used by a library/component in an app with multiple components

I guess that could be ok. Moreover, this saves us some trouble when we'll start collecting telemetry from libraries, preventing them from requiring an environment each when building (think of components in `android-components`, maybe some of them want to be instrumented and we don't really want to have an environment for each of them). Right?

> Having the environment twice is not the best for saving disk space; however that's how the setup with real apps will be too. So i'm not too concerned.

Agreed! However, in general, one wouldn't really need to install the *glean* development environment. App developer would just need to have **one** environment. I think the sample within this repo is a special case, but a corner case as well :)

Reply:
Sorry I missed this comment the first time around.  As it stands, this PR is creating a Python environment for the library itself, and each application that uses it.  This could be unnecessarily large when we start using this in libraries, as @Dexterp37 points out.  But I would argue for merging this as-is with a follow-on bug to address that when we get there.